modules/stochastic_tools/src/utils/GaussianProcess.C

```diff
@@ -0,0 +1,371 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "GaussianProcess.h"
+#include "FEProblemBase.h"
+
+#include <petsctao.h>
+#include <petscdmda.h>
+
+#include "libmesh/petsc_vector.h"
+#include "libmesh/petsc_matrix.h"
+
+#include <cmath>
+
+#include "MooseRandom.h"
+#include "Shuffle.h"
+
+namespace StochasticTools
+{
+
+GaussianProcess::GPOptimizerOptions::GPOptimizerOptions(const bool inp_show_optimization_details,
+                                                        const unsigned int inp_num_iter,
+                                                        const unsigned int inp_batch_size,
+                                                        const Real inp_learning_rate,
+                                                        const Real inp_b1,
+                                                        const Real inp_b2,
+                                                        const Real inp_eps,
+                                                        const Real inp_lambda)
+  : show_optimization_details(inp_show_optimization_details),
+    num_iter(inp_num_iter),
+    batch_size(inp_batch_size),
+    learning_rate(inp_learning_rate),
+    b1(inp_b1),
```

Comment:
Can we please make the names `inp_b1, inp_b2, inp_eps, inp_lambda` more descriptive? Would help someone building upon this class in the future.

Reply:
Done